src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/Namespace/SetAzureEventHubNamespace.cs

```diff
@@ -0,0 +1,82 @@
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+using Microsoft.Azure.Commands.EventHub.Models;
+using System.Collections;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.EventHub.Commands.Namespace
+{
+
+    [Cmdlet(VerbsCommon.Set, EventHubNamespaceVerb), OutputType(typeof(NamespaceAttributes))]
+    public class SetAzureEventHubNamespace : AzureEventHubsCmdletBase
+    {
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+         public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "EventHub Namespace Name.")]
+        [ValidateNotNullOrEmpty]
+        public string NamespaceName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = "EventHub Namespace Location.")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(
+          Position = 3,
+          Mandatory = false,
+          ValueFromPipelineByPropertyName = true,
+          HelpMessage = "Namespace Sku Name.")]
+                [ValidateSet(SKU.Basic,
+          SKU.Standard,
+          SKU.Premium,
+          IgnoreCase = true)]
+        public string SkuName { get; set; }
+
+        [Parameter(
+          Position = 4,
+          Mandatory = false,
+          ValueFromPipelineByPropertyName = true,
+          HelpMessage = "The eventhub throughput units.")]
+        public int? SkuCapacity { get; set; }
+
+        [Parameter(Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 5,
+            HelpMessage = "Disable/Enable Namespace.")]
+        public Management.EventHub.Models.NamespaceState? State { get; set; }
+
+        [Parameter(Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 6,
+            HelpMessage = "Hashtables which represents resource Tags.")]
+        public Hashtable Tags { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            // Update a EventHub namespace 
+            var nsAttribute = Client.UpdateNamespace(ResourceGroupName, NamespaceName, Location, SkuName, SkuCapacity, State, ConvertTagsToDictionary(Tags));
+            WriteObject(nsAttribute);
```

Comment:
You should have presentation types that you use as Output types for your cmdlets. This prevents breaking changes in the underlying management library from being breaking changes in your powershell cmdlets.